src/corehost/cli/hostpolicy/hostpolicy.cpp

```diff
@@ -392,8 +423,58 @@ int corehost_libhost_init(hostpolicy_init_t &hostpolicy_init, const pal::string_
 
 namespace
 {
-    int get_coreclr_delegate(const std::shared_ptr<coreclr_t> &coreclr, coreclr_delegate_type type, void** delegate)
+    int load_runtime_for_context(const context_handle instance)
+    {
+        if (instance == nullptr)
+            return StatusCode::InvalidArgFailure;
+
+        hostpolicy_context_t *context = static_cast<hostpolicy_context_t*>(instance);
```

Comment:
Please introduce a helper to get the `hostpolicy_context_t` from the `context_handle` - including the validation.
It would be nice to have some additional validation on top just a null check. Maybe we can have a marker at the beginning of the struct and check the marker (and explicitly trash it once we close the context). Basically to make it easier to debug cases of usage of context after destruction and so on. The added complexity is very small, and it should help with a class of rather nasty bugs to track down.